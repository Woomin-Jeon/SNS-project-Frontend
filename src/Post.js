/* eslint-disable max-len */
/* eslint-disable object-curly-newline */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';
import { addPost } from './function';
import ShowPost from './ShowPost';

function Post({
  postState,
  setPostState,
  currentUserState,
  setCurrentUserState,
  loginState,
  setLoginState,
  commentState,
  setCommentState,
}) {
  const [temptState, setTemptState] = useState('');
  const { userName, id } = currentUserState;
  const { isLoggedIn } = loginState;

  const setPostTemptContents = (temptState) => {
    setTemptState(temptState);
  };

  const handleAddPost = () => {
    if (temptState.trim()) {
      setPostState(addPost(postState, userName, id, temptState));
      setTemptState('');
    } else {
      alert('내용을 입력해주세요');
    }
  };

  const setIsLoggedInFalse = () => {
    setLoginState({ ...loginState, isLoggedIn: false, temptId: '', temptPw: '' });
    setCurrentUserState({ ...currentUserState, id: '', pw: '', userName: '' });
    alert('로그아웃 되었습니다');
  };

  if (isLoggedIn === false) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <h1>Facebook</h1>
      <button type="button" onClick={setIsLoggedInFalse}>로그아웃</button>
      <div className="addpost">
        <div className="addpost-title">게시물 만들기</div>
        <div className="addpost-notion"><span className="addpost-span">{userName}</span>님, 무슨 생각을 하고 계신가요?</div>
        <div className="addpost-notion">생각을 게시글로 표현 해주세요</div>
        <input className="addpost-inputcontents" type="text" value={temptState} onChange={(e) => setPostTemptContents(e.target.value)} />
        <button className="addpost-out" type="button" onClick={handleAddPost}>게시</button>
      </div>
      <ShowPost
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
        commentState={commentState}
        setCommentState={setCommentState}
      />
    </>
  );
}

export default Post;
