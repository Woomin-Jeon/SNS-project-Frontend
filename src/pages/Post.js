import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import func from '../function';
import Header from '../web_components/Header';
import Advertisement from '../web_components/Advertisement';
import TimeLinePost from '../web_components/TimeLinePost';
import FriendsIndex from '../web_components/FriendsIndex';
import toTop from "../components/toTop";

const callAPI = async (postState, setPostState, loginState, setLoginState) => {
  toTop();
  const { timeLinePosts } = await func.getPosts();
  const { userStore } = await func.getUsers();

  setLoginState({ ...loginState, users: [...userStore] });

  setPostState({
    ...postState,
    post: [...timeLinePosts.reverse()],
  });
};

function Post({
  postState,
  setPostState,
  currentUserState,
  setCurrentUserState,
  loginState,
  setLoginState,
  commentState,
  setCommentState,
  topLevelState,
  setTopLevelState,
  setSearchState,
  socket,
}) {
  const { isLoggedIn, users } = loginState;
  const { id } = currentUserState;

  if (isLoggedIn === false) {
    return <Redirect to="/" />;
  }

  useEffect(() => {
    callAPI(postState, setPostState, loginState, setLoginState);
  }, []);

  return (
    <div>
      {!func.findUserById(users, id)
        ? <h2 className='loading'>페이지를 로딩중입니다...</h2>
        : (
          <div>
            <Header
              loginState={loginState}
              setLoginState={setLoginState}
              currentUserState={currentUserState}
              setCurrentUserState={setCurrentUserState}
              setSearchState={setSearchState}
            />
            <div className="main-timeline">
              <Advertisement />
              <TimeLinePost
                postState={postState}
                setPostState={setPostState}
                currentUserState={currentUserState}
                commentState={commentState}
                setCommentState={setCommentState}
                setTopLevelState={setTopLevelState}
                loginState={loginState}
                setLoginState={setLoginState}
              />
              <FriendsIndex
                loginState={loginState}
                currentUserState={currentUserState}
                topLevelState={topLevelState}
                setTopLevelState={setTopLevelState}
                socket={socket}
              />
            </div>
          </div>
        )}
    </div>
  );
}

export default Post;
