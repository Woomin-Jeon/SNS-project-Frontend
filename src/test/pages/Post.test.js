import Post from "../../pages/Post";
import React from "react";
import {shallow} from 'enzyme';

describe('Post', () => {
  let currentUserState;
  let setCurrentUserState;
  let loginState;
  let setLoginState;
  let commentState;
  let setCommentState;
  let postState;
  let setPostState;
  let setTopLevelState;
  let topLevelState;
  let setSearchState;
  let socket;

  beforeEach(() => {
    postState = { post: [] };
    loginState = { users: [], isLoggedIn: true };
    currentUserState = { id: 'Random_ID' };
  });

  it('renders well', () => {
    const component = shallow(
      <Post
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
        setCurrentUserState={setCurrentUserState}
        loginState={loginState}
        setLoginState={setLoginState}
        commentState={commentState}
        setCommentState={setCommentState}
        topLevelState={topLevelState}
        setTopLevelState={setTopLevelState}
        setSearchState={setSearchState}
        socket={socket}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
