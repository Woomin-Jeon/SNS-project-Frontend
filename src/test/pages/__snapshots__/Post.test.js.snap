// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders well 1`] = `ShallowWrapper {}`;
